scipy/stats/_entropy.py

```diff
@@ -196,7 +257,60 @@ def differential_entropy(
     logs = np.log(
         len(differences) * differences / (2 * window_length),
     )
-    if base is not None:
-        logs /= np.log(base)
 
     return np.mean(logs, axis=0)
+
+
+def _pad_along_last_axis(X, m):
+    """Pad the data for computing the rolling window difference"""
+    # scales a  bit better than method in _vasicek_like_entropy
+    shape = np.array(X.shape)
+    shape[-1] = m
+    Xl = np.broadcast_to(X[..., [0]], shape)  # [0] vs 0 to maintain shape
+    Xr = np.broadcast_to(X[..., [-1]], shape)
+    return np.concatenate((Xl, X, Xr), axis=-1)
+
+
+def _van_es_entropy(X, m):
+    """Compute the van Es estimator as described in [6]"""
+    # No equation number, but referred to as HVE_mn.
+    # Typo: there should be a log within the summation.
+    n = X.shape[-1]
+    term1 = 1/(n-m) * np.sum(np.log((n+1)/m * (X[..., m:] - X[..., :-m])),
+                             axis=-1)
+    k = np.arange(m, n+1)
+    return term1 + np.sum(1/k) + np.log(m) - np.log(n+1)
+
+
+def _ebrahimi_entropy(X, m):
+    """Compute the Ebrahimi estimator as described in [6]"""
+    # No equation number, but referred to as HE_mn
+    n = X.shape[-1]
+    X = _pad_along_last_axis(X, m)
```

Comment:
You're right, but the intent was to avoid modification - even refactoring - of Vasicek where possible to make initial review easier. It does this padding, too, but it doesn't use this function. If you'd like me to refactor Vasicek to make it more consistent with the other methods, I'd be happy to, but for now I left the heart of it alone.

Reply:
The refactoring would be ok for me. But you can leave it for now if you want someone else to review after we iterate.

Reply:
I don't think a second review is needed. If you're willing to merge after reviewing, let me know and I'll refactor Vasicek.

Reply:
I went ahead and did it in ebe65b316487dbbbfec22d386b0f9b64a1b6dbb2.

Reply:
Looks good thanks. It's more readable and consistent with the rest.